src/rewriters/Rewriter.ts

```diff
@@ -56,7 +56,57 @@ abstract class Rewriter {
     return variables;
   }
 
-  public rewriteResponse(response: any, key: string | number): any {
+  /*
+   * Receives the parent object of the matched field with the key of the matched field.
+   * For arrays, the index of the element is also present.
+   */
+  public rewriteResponse(response: any, key: string, index?: number): any {
+    return response;
+  }
+
+  /*
+   * Helper that extracts the element from the response if possible otherwise returns null.
+   */
+  protected extractReponseElement(response: any, key: string, index?: number): any {
+    // Verify the response format
+    let element = null;
+    if (response === null || typeof response !== 'object') return element;
+
+    // Extract the key
+    element = response[key] || null;
+
+    // Extract the position
+    if (Array.isArray(element)) {
+      element = element[index!] || null;
```

Comment:
What is the `!` for?

Reply:
aaah https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator, learned something new! Thanks!